test/extended/images/prune.go

```diff
@@ -175,24 +215,119 @@ func testPruneImages(oc *exutil.CLI, schemaVersion int) {
 		if !strings.Contains(output, layer.Name) {
 			o.Expect(output).NotTo(o.ContainSubstring(layer.Name))
 		}
+		globally, inRepository, err := IsBlobStoredInRegistry(oc, digest.Digest(layer.Name), repoName)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(globally).To(o.BeTrue())
+		o.Expect(inRepository).To(o.BeTrue())
 	}
 
 	confirmSize, err := getRegistryStorageSize(oc)
 	o.Expect(err).NotTo(o.HaveOccurred())
 	g.By(fmt.Sprintf("confirming storage size: sizeOfKeepImage=%d <= sizeAfterPrune=%d < beforePruneSize=%d", imgKeep.DockerImageMetadata.Size, confirmSize, keepSize))
 	o.Expect(confirmSize >= imgKeep.DockerImageMetadata.Size).To(o.BeTrue())
 	o.Expect(confirmSize < keepSize).To(o.BeTrue())
-	g.By(fmt.Sprintf("confirming pruned size: sizeOfPruneImage=%d <= (sizeAfterPrune=%d - sizeBeforePrune=%d)", imgPrune, keepSize, confirmSize))
+	g.By(fmt.Sprintf("confirming pruned size: sizeOfPruneImage=%d <= (sizeAfterPrune=%d - sizeBeforePrune=%d)", imgPrune.DockerImageMetadata.Size, keepSize, confirmSize))
 	o.Expect(imgPrune.DockerImageMetadata.Size <= keepSize-confirmSize).To(o.BeTrue())
 }
 
+func testPruneAllImages(oc *exutil.CLI, setAllImagesToFalse bool, schemaVersion int) {
+	isName := "prune"
+	repository := oc.Namespace() + "/" + isName
+
+	oc.SetOutputDir(exutil.TestContext.OutputDir)
+	outSink := g.GinkgoWriter
+
+	cleanUp := cleanUpContainer{}
+	defer tearDownPruneImagesTest(oc, &cleanUp)
+
+	dClient, err := testutil.NewDockerClient()
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	g.By("build one image using Docker and push it")
+	managedImageName, err := BuildAndPushImageOfSizeWithDocker(oc, dClient, isName, "latest", testImageSize, 2, outSink, true)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	cleanUp.imageNames = append(cleanUp.imageNames, managedImageName)
+	cleanUp.isNames = append(cleanUp.isNames, isName)
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	managedImage, err := oc.AsAdmin().Client().Images().Get(managedImageName, metav1.GetOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	externalImage, blobdgst, err := importImageAndMirrorItsSmallestBlob(oc, externalImageReference, "origin-release:latest")
+	o.Expect(err).NotTo(o.HaveOccurred())
+	cleanUp.imageNames = append(cleanUp.imageNames, externalImage.Name)
+	cleanUp.isNames = append(cleanUp.isNames, "origin-release")
+
+	checkAdminPruneOutput := func(output string, dryRun bool) {
```

Comment:
In this case I'd really prefer it nested. There are too many references to local variables. Plus it doesn't really make sense to call it from anywhere else but this function.